osu.Game/Screens/Play/SoloPlayer.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using osu.Game.Online.API;
+using osu.Game.Online.Rooms;
+using osu.Game.Online.Solo;
+using osu.Game.Scoring;
+
+namespace osu.Game.Screens.Play
+{
+    public class SoloPlayer : SubmittingPlayer
```

Comment:
This is a nit, but ordering of overrides trips me up in this method. Chronologically it goes second, to maybe third, or maybe first.